lesson01/hotel/src/com/andersenhotels/model/service/HotelService.java

```diff
@@ -0,0 +1,107 @@
+package com.andersenhotels.model.service;
+
+import com.andersenhotels.model.Apartment;
+import com.andersenhotels.model.Guest;
+import com.andersenhotels.model.Reservation;
+
+import java.util.*;
+
+/**
+ * The HotelService class manages the registration, reservation,
+ * and listing of apartments in the hotel.
+ */
+public class HotelService {
+    private Map<Integer, Apartment> apartments;
+    private int nextId;
+
+    /**
+     * Constructs a new HotelService instance with an empty apartment registry
+     * and initializes the next available apartment ID.
+     */
+    public HotelService() {
+        apartments = new HashMap<>();
+        nextId = 1;
+    }
+
+    /**
+     * Registers a new apartment with the specified price.
+     *
+     * @param price the price of the apartment.
+     */
+    public void registerApartment(double price) {
+        Apartment apartment = new Apartment(nextId++, price);
+        apartments.put(apartment.getId(), apartment);
+        System.out.println("Apartment registered: " + apartment);
+    }
+
+    /**
+     * Returns the total count of registered apartments.
+     *
+     * @return the number of apartments.
+     */
+    public int getApartmentsCount() {
+        return apartments.size();
+    }
+
+    /**
+     * Reserves an apartment for a guest with the specified name.
+     *
+     * @param id the ID of the apartment to be reserved.
+     * @param guestName the name of the guest reserving the apartment.
+     */
+    public void reserveApartment(int id, String guestName) {
+        Apartment apartment = apartments.get(id);
+        if (apartment != null) {
+            Guest guest = new Guest(guestName);
+            Reservation reservation = new Reservation(apartment, guest);
+            try {
+                reservation.createReservation();
+            } catch (RuntimeException e) {
```

Comment:
try to use a specific exception here) because, as you remember, all Runtimes will be checked here.

Reply:
Done